src/Struct/FindologicProduct.php

```diff
@@ -912,13 +912,12 @@ protected function fetchCategorySeoUrls(CategoryEntity $categoryEntity): SeoUrlC
         $seoUrls = new SeoUrlCollection();
 
         foreach ($categoryEntity->getSeoUrls()->getElements() as $seoUrlEntity) {
-            if ($seoUrlEntity->getSalesChannelId() !== $salesChannelId || $seoUrlEntity->getSalesChannelId() === null) {
-                continue;
+            if ($seoUrlEntity->getSalesChannelId() === $salesChannelId || $seoUrlEntity->getSalesChannelId() === null) {
+                $seoUrls->add($seoUrlEntity);
             }
-            $seoUrls->add($seoUrlEntity);
         }
 
-        return $categoryEntity->getSeoUrls();
+        return $seoUrls;
```

Comment:
Is there a test that covers this newly added logic?

Reply:
Now there is ;)